src/main/java/org/example/EmployeePaymentStrategyFactory.java

```diff
@@ -0,0 +1,20 @@
+package org.example;
+
+import org.example.data.Employee;
+import org.example.exceptions.InvalidEmployeeTypeException;
+import org.example.paymentstrategies.CommissionedEmployeePaymentStrategy;
+import org.example.paymentstrategies.EmployeePaymentStrategy;
+import org.example.paymentstrategies.HourlyEmployeePaymentStrategy;
+import org.example.paymentstrategies.SalariedEmployeePaymentStrategy;
+
+public class EmployeePaymentStrategyFactory {
+  EmployeePaymentStrategy createPaymentStrategy(Employee employee) throws InvalidEmployeeTypeException {
+
+    return switch (employee.getType()) {
```

Comment:
still switch exists - any time new type of employee appear - you need to create a new row in switch
https://www.informit.com/articles/article.aspx?p=2301788&seqNum=2

Reply:
Yes, but if the example provided by the link has to decide which writer to use, it will also need switch.
If I delete the EmployeePaymentStrategyFactory from my example and just leave different interfaces for use, then my solution will have no switch.